libs/core/src/components/grid/grid.ts

```diff
@@ -0,0 +1,126 @@
+import { LitElement, unsafeCSS } from 'lit'
+import { property } from 'lit/decorators.js'
+import {
+  gdsCustomElement,
+  html,
+} from '../../utils/helpers/custom-element-scoping'
+
+import { tokens } from '../../tokens.style'
+import grid from './grid.style.css'
+const gridCSS = unsafeCSS(grid)
+
+type GridSizes = 'none' | 'xs' | 's' | 'm' | 'l' | 'xl' | '2xl' | '3xl'
+type GridGap = GridSizes
+type GridPadding = GridSizes
+
+@gdsCustomElement('gds-grid')
+export class GdsGrid extends LitElement {
+  static styles = [tokens, gridCSS]
+
+  static shadowRootOptions: ShadowRootInit = {
+    mode: 'open',
+    delegatesFocus: true,
+  }
+
+  @property({ attribute: 'columns', type: String })
+  columns?: string
+
+  @property({ attribute: 'gap', type: String })
+  gap?: GridGap
+
+  @property({ attribute: 'padding', type: String })
+  padding?: GridPadding
+
+  @property({ attribute: 'fluid', type: String })
+  fluid?: boolean
+
+  updated(changedProperties: Map<string | number | symbol, unknown>) {
```

Comment:
Instead of using `updated` and checking for each attribute name here, you could also use the `@watch` decorator.

Example:
```ts
@watch('columns')
private _handleColumnsChanged() {
  // ... code for columns changed
}
```